src/testHelpers.ts

```diff
@@ -26,6 +27,17 @@ export function initTestHelpers() {
   if (typeof document !== 'undefined' && typeof afterEach === 'function') {
     afterEach(cleanup);
   }
+
+  if (typeof jest !== 'undefined') {
```

Comment:
By moving this into `testHelpers` we are making usage of `initTestHelpers` mandatory for the `getPage` to work correctly. This might be a bit annoying and I'm sure some people might forget to do it (or intentionally avoid it) and open issues. What do you think?

Just curious: what happens if we move this into somewhere early in `getPage` -- does it work?

Reply:
This was exactly my concern.
For the sake of our emotion setup, this would also work if we kept non-isolaated modules Jest loop in `utils` (executed at module loading time). I can't guarantee this would work in any context, though.

Since we're Jest mocking things, I though the best place was in a `beforeAll` callback. But `testHelpers` setup becomes mandatory, then.

I start to suspect we stumbled upon another Jest bug.

Reply:
So yeah it's a decision we need to make. If we move it `initTestHelpers` I think we need to improve the DX as well by auto initing `testHelpers`.

I think its very off-putting when you want to try integrating a new library into a complex project and you see some errors. I think it should just work out of the box.

For advanced users that want to opt-out from this behaviour and control it themselves we can provide exactly the same approach as `@testing-library/react`.

Reply:
> I think we need to improve the DX as well by auto initing testHelpers.

 You mean avoid letting users to drop these lines in their `setupFilesAfterEnv` files?
 
 ```
import { initTestHelpers } from 'next-page-tester';
initTestHelpers();
 ```

In order to make a step further we should hook into Jest and programmatically configure `setupFilesAfterEnv`. I've never tried. If possible, non-Jest (phantomatic?) users would still have to do it manually.

Or you're more thinking about detecting that proper env configuration is missing and alert users?

Reply:
Yes they would be able to omit those lines, but more importantly -- new users wont see errors in case they forget it. Ofcourse they would go to the repo and in the `README.md` read they need to use it, but we can them (and ourselves -- by receiving less issues) a favour if we wire everything up ourselves.

So all that is needed is something like this:
- Rename current `index.ts` -> `pure.ts`
- Add a new `index.ts` with content similar to this


```ts
import { initTestHelpers } from 'next-page-tester'
export * from './pure'

if(!process.env.NEXT_PAGE_TESTER_SKIP_TEST_INIT) {
    initTestHelpers();
}
```

Reply:
I see now. I wasn't aware of such a trick in testing-library. Would it somehow also work for `setupFilesAfterEnv`? I'm totally open to such a solution if we found a proper way to test it.

Off topic: I'm a bit concerned about how we now replace/filter runtime (runtime only) env vars. We get closer to what's expected in a Next.js runtime routine but when in client environment (which we consider the base one) we're practically filtering everything out. Let's see if it break things :)

Reply:
Not sure what you mean with `setupFilesAfterEnv`. This is just jest's specific hook to initialise some stuff. 

Whats great about the trick I've proposed and testing-library is using is that it works by default with every testing framework that supports `afterEach` callback (most of them do).

No need to do anything with `setupFilesAfterEnv`.

When user imports `getPage`:

```js
import { getPage } from 'next-page-tester'
```

Module will get executed and setup the `afterEach` callbacks and mocks.

Reply:
Oki, let's give it a try. I'll merge this PR and maybe release v0.21.0 so that we can better focus on what's next. The great thing about living in v0 😄